### PR TITLE
Update complexity controls.

### PR DESCRIPTION
**Pull request by davidlehn**
- Add `maxWorkFactor` parameter.
- Update tests and docs.
- Other misc fixes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,13 @@
 - Add `signal` option to allow use of an `AbortSignal` for complexity control.
   Enables the algorithm to abort after a timeout, manual abort, or other
   condition.
+- Add `maxWorkFactor` to calculate a deep iteration limit based on the number
+  of non-unique blank nodes. This defaults to `1` for roughly O(n) behavior and
+  will handle common graphs. It must be adjusted to higher values if there is a
+  need to process graphs with complex blank nodes or other "poison" graphs. It
+  is recommeded to use this paramter instead of `maxDeepIterations` directly.
```
Comment (dlongley) [inline CHANGELOG.md:20]:
```suggestion
  is recommended to use this parameter instead of `maxDeepIterations` directly.
```

Comment (dlongley) [reply CHANGELOG.md:20]: Should state that `maxDeepIterations` is provided, then `maxWorkFactor` will be ignored.

File: README.md [hunk 2]
```diff
@@ -131,15 +136,37 @@ can be used in a number of ways:
...
+  to `Infinity` and any other value will override `maxWorkFactor`.
+
+### Usage
 
 In practice, callers must balance system load, concurrent processing, expected
 input size and complexity, and other factors to determine which complexity
-controls to use. This library defaults to infinite `maxDeepIterations` and a 1
-second timeout, and these should be adjusted as needed.
+controls to use. This library defaults to a `maxWorkFactor` of `1` and no
+timeout signal. These should be adjusted as needed.
```
Comment (dlongley) [inline README.md:169]:
```suggestion
timeout signal. These can be adjusted as needed.
```

File: lib/URDNA2015.js [hunk 3]
```diff
@@ -99,6 +100,24 @@ module.exports = class URDNA2015 {
       // 5.4.5) Set simple to true.
     }
 
+    if(this.maxDeepIterations < 0) {
+      // calculate maxDeepIterations if not explicit
+      if(this.maxWorkFactor === 0) {
+        // use 0 default
```
Comment (dlongley) [inline lib/URDNA2015.js:106]: I found this comment a bit confusing ... `maxDeepIterations` is less than `0` right now ... so should it be set to `0`? What does "use 0 default" mean?

It looks like `this.remainingDeepIterations` will be set to `-1` in that case ... maybe the code on lines 155-119 that sets it should be moved inside of the `this.maxDeepIterations < 0` conditional ... either that or set `maxDeepIterations` to `0` explicitly?

Comment (davidlehn) [reply lib/URDNA2015.js]: A refactoring broke the logic here.  Fixed it and added some tests.

**Pull request merged**